### PR TITLE
added SystemSofastFixed

### PR DESCRIPTION
**Pull request by braden6521**
- Renamed old SystemSofastFixed to PatternSofastFixed (analog of SofastFringe `Fringes.py` class)
- Created new SystemSofastFixed (analog of SystemSofastFringe)
- Pulled all of these together into a Sofast command line interface (/contrib/app/sofast/)

### CODE REVIEW

File: opencsp/app/sofast/test/ImageAcquisition_no_camera.py [hunk 1]
```diff
@@ -34,6 +34,7 @@ def instance_matches(self, possible_matches: list[ImageAcquisitionAbstract]) ->
         return False
 
     def get_frame(self) -> np.ndarray:
+        time.sleep(self._shutter / 1e6)
```
Comment (bbean23) [inline opencsp/app/sofast/test/ImageAcquisition_no_camera.py:37]: Huh, didn't realize this wasn't here. Nifty!

**Review by bbean23 — CHANGES_REQUESTED** [review]
Again, I have just a few comments. Please address them and I'll approve.

File: opencsp/app/sofast/lib/SystemSofastFixed.py [hunk 2]
```diff
@@ -1,122 +1,186 @@
...
 
 class SystemSofastFixed:
-    """Class that holds parameters for displaying a Fixed Pattern for use
-    in fixed pattern deflectometry.
-    """
+    """Class that orchestrates the running of a SofastFixed system"""
+
+    def __init__(self, image_acquisition: ImageAcquisitionAbstract) -> 'SystemSofastFixed':
+        """
+        Instantiates SystemSofastFringe class.
```
[the remark this answers is not in the document]
Comment (braden6521) [reply opencsp/app/sofast/lib/SystemSofastFixed.py]: Fixed.

File: contrib/app/sofast/sofast_command_line_interface.py [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+    # Load common data
+    facet_definition = DefinitionFacet.load_from_json(file_facet_definition_json)
+    spatial_orientation = SpatialOrientation.load_from_hdf(file_spatial_orientation)
+    display = DisplayShape.load_from_hdf(file_display)
+    camera = Camera.load_from_hdf(file_camera)
+    fixed_pattern_dot_locs = DotLocationsFixedPattern.load_from_hdf(file_dot_locs)
+
+    periods_x = [0.9, 4.0, 16.0, 64.0]
+    periods_y = [0.9, 4.0, 16.0, 64.0]
+    fringes = Fringes(periods_x, periods_y)
```
[the remark this answers is not in the document]
Comment (braden6521) [reply contrib/app/sofast/sofast_command_line_interface.py]: Nice; changed!

File: contrib/app/sofast/sofast_command_line_interface.py [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+def _timestamp() -> str:
+    return dt.datetime.now().isoformat().replace(":", "_")
+
+
+def main():
+    """Main system_fringe runner"""
+    # Define save direcory and set up logger
+    dir_save = abspath(join(dirname(__file__), '../../../../sofast_cli'))
+    dir_log = join(dir_save, 'logs')
+    lt.logger(join(dir_log, f'log_{_timestamp():s}.txt'), lt.log.INFO)
```
[the remark this answers is not in the document]
Comment (braden6521) [reply contrib/app/sofast/sofast_command_line_interface.py]: Done.

File: contrib/app/sofast/sofast_command_line_interface.py [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+
+
+def _timestamp() -> str:
+    return dt.datetime.now().isoformat().replace(":", "_")
+
+
+def main():
+    """Main system_fringe runner"""
+    # Define save direcory and set up logger
+    dir_save = abspath(join(dirname(__file__), '../../../../sofast_cli'))
```
[the remark this answers is not in the document]
Comment (braden6521) [reply contrib/app/sofast/sofast_command_line_interface.py]: All inputs are now parameterized.

**Comment by braden6521** [comment]
@bbean23, thanks for comments! I believed I have addressed them all.

**Review by bbean23 — APPROVED** [review]
Looks good. Approved!

**Pull request merged**